internal/db/collection_index.go

```diff
@@ -395,6 +395,114 @@ func (c *collection) GetIndexes(context.Context) ([]client.IndexDescription, err
 	return c.Version().Indexes, nil
 }
 
+// CreateEncryptedIndex creates a new encrypted index on the collection.
+// `EncryptedIndexCreateRequest` contains the description of the index to be created.
+func (c *collection) CreateEncryptedIndex(
+	ctx context.Context,
+	createRequest client.EncryptedIndexCreateRequest,
+) (client.EncryptedIndexDescription, error) {
+	ctx, span := tracer.Start(ctx)
+	defer span.End()
+
+	ctx, txn, err := ensureContextTxn(ctx, c.db, false)
+	if err != nil {
+		return client.EncryptedIndexDescription{}, err
+	}
+	defer txn.Discard(ctx)
+
+	index, err := c.createEncryptedIndex(ctx, createRequest)
+	if err != nil {
+		return client.EncryptedIndexDescription{}, err
+	}
+	return index, txn.Commit(ctx)
+}
+
+func (c *collection) createEncryptedIndex(
+	ctx context.Context,
+	createRequest client.EncryptedIndexCreateRequest,
+) (client.EncryptedIndexDescription, error) {
+	desc, err := processCreateEncryptedIndexRequest(c.Version(), createRequest)
+	if err != nil {
+		return client.EncryptedIndexDescription{}, err
+	}
+
+	c.def.EncryptedIndexes = append(c.def.EncryptedIndexes, desc)
+
+	err = description.SaveCollection(ctx, c.def)
+	if err != nil {
+		c.def.EncryptedIndexes = c.def.EncryptedIndexes[:len(c.def.EncryptedIndexes)-1]
+		return client.EncryptedIndexDescription{}, err
+	}
+
+	err = c.db.loadSchema(ctx)
+	if err != nil {
+		return client.EncryptedIndexDescription{}, err
+	}
+
+	return c.def.EncryptedIndexes[len(c.def.EncryptedIndexes)-1], nil
+}
+
+// GetEncryptedIndexes returns all the encrypted indexes that exist on the collection.
+func (c *collection) GetEncryptedIndexes(ctx context.Context) ([]client.EncryptedIndexDescription, error) {
+	return c.Version().EncryptedIndexes, nil
+}
```

Comment:
suggestion: Probably want to stick with the `list` naming like you did in client cli commands for example, due to : https://github.com/sourcenetwork/defradb/issues/3881

```suggestion
// ListEncryptedIndexes returns all the encrypted indexes that exist on the collection.
func (c *collection) ListEncryptedIndexes(ctx context.Context) ([]client.EncryptedIndexDescription, error) {
	return c.Version().EncryptedIndexes, nil
}
```

Reply:
done